src/Operation/EstimatedDocumentCount.php

```diff
@@ -73,9 +83,24 @@ public function __construct($databaseName, $collectionName, array $options = [])
     {
         $this->databaseName = (string) $databaseName;
         $this->collectionName = (string) $collectionName;
-        $this->options = array_intersect_key($options, ['maxTimeMS' => 1, 'readConcern' => 1, 'readPreference' => 1, 'session' => 1]);
 
-        $this->count = $this->createCount();
+        if (isset($options['maxTimeMS']) && ! is_integer($options['maxTimeMS'])) {
```

Comment:
I decided that checking options for `estimatedDocumentCount` here specifically makes sense now that we no longer know which command (count vs. aggregate) is being executed.

Reply:
Both Count and Aggregate do the same checks for these options, so does this change anything?

Perhaps this is needed to ensure we raise these errors in the constructor, since construction of the other operation is now deferred to `execute()`? I suppose the only alternative would be to create both a Count and Aggregate operation up front and decide which is used later.

Reply:
Yes, that is the main difference. I considered instantiating both operation objects up front, but avoided this as one object may throw an exception during initialisation even though it wouldn't be used later on.